src/linear_algebra/special_linear_group.lean

```diff
@@ -0,0 +1,133 @@
+/-
+  Copyright (c) 2020 Anne Baanen. All rights reserved.
+  Released under Apache 2.0 license as described in the file LICENSE.
+  Author: Anne Baanen.
+
+  The Special Linear group $SL(n, R)$
+-/
+import linear_algebra.basic
+import linear_algebra.matrix
+import linear_algebra.nonsingular_inverse
+import tactic.norm_cast
+
+/-!
+# The Special Linear group $SL(n, R)$
+
+This file defines the elements of the Special Linear group `special_linear_group n R`,
+also written `SL(n, R)` or `SLₙ(R)`, consisting of all `n` by `n` `R`-matrices with
+determinant `1`.  In addition, we define the group structure on `special_linear_group n R`
+and the embedding into the general linear group `general_linear_group R (n → R)`
+(i.e. `GL(n, R)` or `GLₙ(R)`).
+
+## Main definitions
+
+ * `matrix.special_linear_group` is the set of matrices with determinant 1
```

Comment:
```suggestion
 * `matrix.special_linear_group` is the type of matrices with determinant 1
```
Also, you could maybe add here the parameters that one should feed to this function, i.e., `n R`.